src/lib.rs

```diff
@@ -301,7 +295,7 @@ pub struct EGraph {
 /// it has an exclusive access to the e-graph.
 pub trait UserDefinedCommand: Send + Sync {
     /// Run the command with the given arguments.
-    fn update(&self, egraph: &mut EGraph, args: &[Expr]) -> Result<(), Error>;
+    fn update(&self, egraph: &mut EGraph, args: &[Expr]) -> Result<Option<CommandOutput>, Error>;
```

Comment:
if `update` returns an optional `CommandOutput`, then we need to have a CommandOutput variant for custom user outputs.